lib/json_api_resource/resource.rb

```diff
@@ -46,7 +46,11 @@ def persisted?
       !new_record?
     end
 
-    def attributes=(attr = {})
+    def attributes=(attr = nil)
+      # The difference between doing this vs as the default parameter
+      # is that you can try to set attributes to nil without an exception this way
+      attr ||= HashWithIndifferentAccess.new
```

Comment:
So, for sure, attributes are not nil.  Even if you pass in nil, we change it to hash.